db/document.go

```diff
@@ -1151,14 +1209,20 @@ func (doc *Document) UnmarshalWithXattr(ctx context.Context, data []byte, xdata
 	case DocUnmarshalNoHistory:
 		// Unmarshal sync metadata only, excluding history
 		doc.SyncData = SyncData{}
-		unmarshalErr := base.JSONUnmarshal(xdata, &doc.SyncData)
+		unmarshalErr := base.JSONUnmarshal(syncXattrData, &doc.SyncData)
 		if unmarshalErr != nil {
 			return pkgerrors.WithStack(base.RedactErrorf("Failed to UnmarshalWithXattr() doc with id: %s (DocUnmarshalNoHistory).  Error: %v", base.UD(doc.ID), unmarshalErr))
 		}
+		if hlvXattrData != nil {
+			err := base.JSONUnmarshal(hlvXattrData, &doc.SyncData.HLV)
+			if err != nil {
+				return pkgerrors.WithStack(base.RedactErrorf("Failed to unmarshal HLV during UnmarshalWithXattr() doc with id: %s (DocUnmarshalAll/Sync).  Error: %v", base.UD(doc.ID), err))
```

Comment:
```suggestion
				return pkgerrors.WithStack(base.RedactErrorf("Failed to unmarshal HLV during UnmarshalWithXattr() doc with id: %s (DocUnmarshalNoHistory).  Error: %v", base.UD(doc.ID), err))
```

I'd be tempted to do something like define:

```
type DocumentUnmarshalLevel uint8

func (l DocumentUnmarshalLevel) String() {
	switch l {
	case DocUnmarshalLevel:
		return "DocUnmarshalAll"     
	case DocUnmarshalSync:
		return "DocUnmarshalSync"
	case DocUnmarshalNoHistory:
		return "DocUnmarshalNoHistory"
	case DocUnmarshalHistory:
		return "DocUnmarshalHistory"
	case DocUnmarshalRev:
		return "DocUnmarshalRev"
	case DocUnmarshalCAS:
		return "DocUnmarshalCAS"
	case DocUnmarshalNone:
		return "DocUmarshalNone"
	}
	return "DocumentUnmarshalLevel %d", l)
)

Reply:
That's a reasonable idea, would probably want to do that in a separate PR.